package.json

```diff
@@ -1,52 +1,54 @@
 {
-  "name": "serialport-gsm",
-  "version": "3.4.2",
-  "description": "Serialport implementation for GSM commands (send text/flash messages, read, etc.)",
-  "main": "lib/",
-  "directories": {
-    "lib": "lib"
-  },
-  "dependencies": {
-    "node-pdu": "^1.0.15",
-    "serialport": "^8.0.8"
-  },
-  "devDependencies": {
-    "nodemon": "^1.18.9"
-  },
-  "scripts": {
-    "start": "nodemon test.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
-  },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/zabsalahid/serialport-gsm.git"
-  },
-  "keywords": [
-    "gsm",
-    "at",
-    "commands",
-    "modem",
-    "sms",
-    "iot",
-    "serial port",
-    "serialport",
-    "serial",
-    "com",
-    "port",
-    "flash message",
-    "class 0"
-  ],
-  "author": "Zabdiel G. Salahid <zabsalahid@gmail.com> (https://github.com/zabsalahid)",
-  "contributors": [
-    {
-      "name": "Paolo Brainard M. Reyes",
-      "email": "xreyes.paolox@gmail.com",
-      "url": "http://360logix.com"
-    }
-  ],
-  "license": "MIT",
-  "bugs": {
-    "url": "https://github.com/zabsalahid/serialport-gsm/issues"
-  },
-  "homepage": "https://github.com/zabsalahid/serialport-gsm#readme"
+	"name": "serialport-gsm",
+	"version": "3.4.2",
+	"description": "Serialport implementation for GSM commands (send text/flash messages, read, etc.)",
+	"main": "./dist/index.js",
+	"types": "./dist/index.d.ts",
+	"dependencies": {
+		"node-pdu": "https://github.com/KillerJulian/node-pdu",
+		"serialport": "10.4.0"
```

Comment:
why only static dependencies?

Reply:
It is better to lock the dependencies, otherwise it may cause problems under different versions

Reply:
In general, IF developers do not use semantic versioning you are right. And in fact you also would be safe for "npm new version breaks it by intent" as it happend with the color library ... BUT This way of versioning ( at least in my eyes) mean that you **need to** problish new versions on any security relevant library updates because npm will not do that by the normal semantic versioning things ... 

My result so far was that it casuses issues once a year - and I maintain a loooot of projects ... so it is not really a practical issue and releasing updates on any dependency update is more effort. (and in fact you are not more secure because 99% of all deps use ^ and so it is in fact not safer because sub-deos will still be different :-)

Reply:
Due to the changes in the next version the dependencies must be locked

Reply:
Then you cold release a scoped package under your user to npm ... this is much better then requirung all users to have git installs that want to use this package (osme do not have that) and with a scoped package you have the big benefit of proper versioning. r ztry to contact the package author also via email or such ato forcie or officially take the package over